### PR TITLE
[PATH]  CAMotics simulation.  First Draft

### PR DESCRIPTION
**Pull request by sliptonic**
This is a first attempt at a CAMotics integration.  It adds a new tool to the toolbar if CAMotics is found on the system.
This requires a very late version of CAMotics (1.2.2) 

When a job is selected, the tool becomes active.  clicking the button opens the simulation panel.  this silently postprocesses the job in the background and builds the camotics project including tools and stock.  It will estimate the runtime of the job.  The user can then click the 'run' button in task panel to generate the simulated output mesh.

The user can also select an intermediate point in the list of operations.  Doing so will update the estimate to reflect runtime to that point.  Running the simulation will produce the shape as of that point.

Finally, the user can open the camotics UI directly.  This will require a full-postprocessing of the job and also require the user to specify a camotics project file name.

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [ ]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [ ]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [ ]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [ ]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [ ]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [ ]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [ ]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the [FreeCAD bug tracker issue number](https://freecadweb.org/wiki/tracker#GitHub_and_MantisBT) in case a particular commit solves or is related to an existing issue on the tracker. Ex: `Draft: fix typos - fixes #0004805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.19 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=34586).

---

### CODE REVIEW

**Comment by snelweg** [comment]
Wow, this is so much more then expected. This is a truly worthwhile extension of FreeCAD and a confirmation that CAMotics is one of the best (maybe the best) cam simulation software available.
Nice to see this development.

**Comment by luzpaz** [comment]
Issues with windows on TravisCI:
* https://travis-ci.org/github/FreeCAD/FreeCAD/jobs/764437129#L251-L253
* https://travis-ci.org/github/FreeCAD/FreeCAD/jobs/764437129#L303-L312

**Comment by chennes** [comment]
That Travis build is using an old libpack download link -- rebasing from the current master should fix that.

**Comment by luzpaz** [comment]
@sliptonic good news, travisci exited successfully. congrats

File: src/Mod/Path/PathScripts/PathCamoticsGui.py [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+
+import FreeCAD
+import FreeCADGui
+import PathScripts.PathLog as PathLog
+# from pivy import coin
+# from itertools import cycle
+import json
+import Mesh
+import time
+import camotics
```
Comment (dubstar-04) [inline src/Mod/Path/PathScripts/PathCamoticsGui.py:31]: Where does the camotics module come from?

**Comment by dubstar-04** [comment]
missing  PathScripts/PathCamoticsGui.py from  path cmakelists.

**Comment by youyizhang** [comment]
Hey, I build the camotics 1.2.2 in Ubuntu 20.04 and clone this link: https://github.com/sliptonic/FreeCAD/tree/feature/camotics
But my toolbar did not have the camotics icon.

**Comment by luzpaz** [comment]
@[youyizhang](https://github.com/youyizhang) ping

**Pull request merged**